### ISSUE TITLE
io.realm.exceptions.RealmException: Primary key constraint broken. Value already exists: 0

### ISSUE THREAD

**Issue by cmonfortep** [realm/realm-java#1013]
I'm have an issue trying to insert multiple objects of this type.

```
public class realmTestEntity extends RealmObject {
    @PrimaryKey
    private int code;

    //other attributes

//getters and setters
}
```

If I create a list of realmTestEntity with codes from 0 to 12, I have an error of "Primary key constraint broken. Value already exists: 0" when I try to insert the objects. But if my List has objects with codes from 1 to 12, everything works fine.

I have tested the code passing a List or every object with a loop, always with the same exception if there is an object with "code" set to 0, I can't insert more objects. The same code works if the list doesn't have an object with "code" set to 0.

Is there any constraint setting a primary key to 0?

**Comment by emanuelez** [comment]
0 is the default value for int fields, so if you have a RealmObject with 0 as the value for your `code` field, it means that `realm.createObject(RealmTestEntity.class)` will fail with the error you mentioned.

**Comment by cmonfortep** [comment]
The thing is that I can insert an object with 0 as the value for "code" without exception, but then I can not insert more objects in that table (with different values for "code" field).

**Comment by emanuelez** [comment]
Exactly. That's expected behavior. The trick is to never add an object with the default value for a primary key field since that will unable you to create any other Realm object

**Comment by cmonfortep** [comment]
Ok thank you, I understand what you want to say.

I think this behavior should be documented.

**Comment by emanuelez** [comment]
It is, in the second paragraph here http://realm.io/docs/java/0.80.0/#primary-keys
Maybe we should add it to the javadocs as well

**Comment by cmonfortep** [comment]
Yes, and that's why I opened an issue.

##### In order to avoid conflicts with another object with the same primary key, it is suggested to create a standalone object, set the values of the fields, and then copy the object to Realm, using the copyToRealm() method.

I've always tried to insert the objects with ```copyToRealm()``` and I got the exception. If I add an object with code=0, then I can not create more objects even if I'm using copyToRealm. Reading the docs I understand that I can create object if they are already populated using this method.

**Comment by cmelchior** [comment]
Hi @cmonfortep 
Yes, that is a bug in copyToRealm(). We have merged a fix to master: https://github.com/realm/realm-java/pull/995, but it hasn't been properly released yet, until then you can use the our SNAPSHOT release to test it: https://github.com/realm/realm-java/issues/936#issuecomment-77676494. Latest is 80.1-SNAPSHOT.

**Comment by vijaemanlapaz** [comment]
This still occurs in 0.81.1

**Comment by Rexota** [comment]
Also same problem in realm-0.82.1 .

**Comment by BrunoVieiraAndrade** [comment]
Also same problem in realm-0.87.2

**Comment by johan-steffens** [comment]
This is still happening in realm-0.88.3

**Comment by Shraddha13Gupta** [comment]
this issue should be documented and get rid off!

**Comment by Zhuinden** [comment]
@Shraddha13Gupta this issue was fixed in 2.0+ and avoidable by using `createObject(clazz, primaryKeyValue)` instead of `createObject(clazz)`
